eclair-core/src/main/scala/fr/acinq/eclair/router/Router.scala

```diff
@@ -162,19 +162,25 @@ class Router(val nodeParams: NodeParams, watcher: ActorRef, initialized: Option[
       log.info("reinstating shortChannelId={} from nodeId={}", shortChannelId, nodeId)
       stay using d.copy(excludedChannels = d.excludedChannels - desc)
 
-    case Event(Symbol("nodes"), d) =>
+    case Event(GetNodes, d) =>
       sender ! d.nodes.values
       stay
 
-    case Event(Symbol("channels"), d) =>
+    case Event(GetLocalChannels, d) =>
+      val scids = d.graph.getIncomingEdgesOf(nodeParams.nodeId).map(_.desc.shortChannelId)
+      val localChannels = scids.flatMap(scid => d.channels.get(scid).map(c => LocalChannel(Right(c))).orElse(d.privateChannels.get(scid).map(c => LocalChannel(Left(c)))))
```

Comment:
Slightly shorter version. I tried linebreaks, but it wasn't much more readable. We could save one iteration by removing the `.map(_.desc.shortChannelId)`.

```suggestion
      val localChannels = scids.flatMap(scid => d.channels.get(scid).map(Right(_)).orElse(d.privateChannels.get(scid).map(Left(_)).map(LocalChannel)))
```